src/documents/serialisers.py

```diff
@@ -472,6 +476,61 @@ def validate(self, data):
             URLValidator()(data["value"])
         return data
 
+    def reflect_doclinks(
+        self,
+        document: Document,
+        field: CustomField,
+        target_doc_ids: list[int],
+    ):
+        """
+        Add or remove 'symmetrical' links to `document` on all `target_doc_ids`
+        """
+        # Check if any documents are going to be removed from the current list of links and remove the symmetrical links
+        current_field_instance = CustomFieldInstance.objects.filter(
+            field=field,
+            document=document,
+        ).first()
+        if current_field_instance is not None:
+            for doc_id in current_field_instance.value:
+                if doc_id not in target_doc_ids:
+                    self.remove_doclink(document, field, doc_id)
+
+        for target_doc_id in target_doc_ids:
+            target_doc_field_instance = CustomFieldInstance.objects.filter(
+                field=field,
+                document_id=target_doc_id,
+            ).first()
+            if target_doc_field_instance is not None:
+                if document.id not in target_doc_field_instance.value:
+                    target_doc_field_instance.value_document_ids.append(document.id)
+                    target_doc_field_instance.save()
+            else:
+                CustomFieldInstance.objects.update_or_create(
+                    document_id=target_doc_id,
+                    field=field,
+                    defaults={"value_document_ids": [document.id]},
+                )
```

Comment:
I would suggest getting rid of db queries in the loop (n+1 problem). We may consider to reduce number of queries using something like:
```suggestion
        custom_field_instances_to_add = []
        custom_field_instances = {
            custom_field.document_id: custom_field
            for custom_field in CustomFieldInstance.objects.filter(
                field=field,
                document_id__in=target_doc_ids,
            )
        }
        for target_doc_id in target_doc_ids:
            target_doc_field_instance = custom_field_instances.get(target_doc_id)
            if target_doc_field_instance is not None:
                if document.id not in target_doc_field_instance.value:
                    target_doc_field_instance.value_document_ids.append(document.id)
                    target_doc_field_instance.save(update_fields=["value_document_ids"])
            else:
                custom_field_instances_to_add.append(
                    CustomFieldInstance(
                        document_id=target_doc_id,
                        field=field,
                        value_document_ids=[document.id],
                    ),
                )
        CustomFieldInstance.objects.bulk_create(custom_field_instances_to_add)
```

Reply:
Thanks, had that thought while writing this but never went back to it. I’ll take a look

Reply:
Thanks for the suggestion, I wasnt really familiar with the Django bulk operations either, I think those are gonna come in handy again